cmd/libs/go2idl/types/types.go

```diff
@@ -285,9 +285,32 @@ func (t *Type) String() string {
 	return t.Name.String()
 }
 
-// IsAssignable returns whether the type is assignable.
+// IsPrimitive returns whether the type is a built-in type or is an alias to a
+// built-in type.  For example: strings and aliases of strings are primitives,
+// structs are not.
+func (t *Type) IsPrimitive() bool {
+	if t.Kind == Builtin || (t.Kind == Alias && t.Underlying.Kind == Builtin) {
+		return true
+	}
+	return false
+}
+
+// IsAssignable returns whether the type is deep-assignable.  For example,
+// slices and maps and pointers are shallow copies, but ints and strings are
+// complete.
 func (t *Type) IsAssignable() bool {
-	return t.Kind == Builtin || (t.Kind == Alias && t.Underlying.Kind == Builtin)
+	if t.IsPrimitive() {
+		return true
+	}
+	if t.Kind == Struct {
+		for _, m := range t.Members {
+			if !m.Type.IsAssignable() {
```

Comment:
No.  Go will always copy structs.  In your example:

```
x1 := randomX()
x2 := x1 // full deep copy
```

This is pretty obvious once you think about it, and easily verified.
Or did I miss the point you were making?